gamestonk_terminal/options/volume_view.py

```diff
@@ -0,0 +1,463 @@
+        df_calls["dv"] = df_calls["dv"].apply(lambda x: max(0, x))
+        df_calls["dv"] = abs(df_calls["dv"] * df_calls["volume"])
+
+    if not op_puts.empty:
+        # Process Puts Data
+        df_puts = op_puts.pivot_table(
+            index="strike", values=["volume", "openInterest"], aggfunc="sum"
+        ).reindex()
+        df_puts["strike"] = df_puts.index
+        df_puts["type"] = "puts"
+        df_puts["openInterest"] = df_puts["openInterest"]
+        df_puts["volume"] = -df_puts["volume"]
+        df_puts["openInterest"] = -df_puts["openInterest"]
+        df_puts["oi+v"] = df_puts["openInterest"] + df_puts["volume"]
+        df_puts["spot"] = round(last_adj_close_price, 2)
+        df_puts["dv"] = df_puts.index - last_adj_close_price
+        df_puts["dv"] = df_puts["dv"].apply(lambda x: max(0, x))
+        df_puts["dv"] = abs(df_puts["dv"] * df_puts["volume"])
+
+    if not op_calls.empty and not op_puts.empty:
+        # Get max pain
+        df_opt = pd.merge(df_calls, df_puts, left_index=True, right_index=True)
+        df_opt["dv"] = round(df_opt["dv_x"] + df_opt["dv_y"], 2)
+        max_pain = df_opt["dv"].idxmax()
+
+    return df_calls, df_puts, max_pain
+
+
+def plot_calls_volume_open_interest(
+    other_args: List[str],
+    ticker: str,
+    exp_date: str,
+    last_adj_close_price: float,
+    op_calls: pd.DataFrame,
+):
+    """Options volume and open interest
+
+    Parameters
+    ----------
+    other_args : List[str]
+        Command line arguments to be processed with argparse
+    ticker : str
+        Main ticker to compare income
+    exp_date : str
+        Expiry date of the option
+    last_adj_close_price: float
+        Last adjusted closing price
+    op_calls: pd.DataFrame
+        Option data calls
+    """
+    parser = argparse.ArgumentParser(
+        add_help=False,
+        prog="vcalls",
+        description="""
+            Plots Calls Volume + Open Interest. [Source: Yahoo Finance]
+        """,
+    )
+    parser.add_argument(
+        "-m",
+        "--min",
+        dest="min_sp",
+        type=check_non_negative,
+        default=-1,
+        help="minimum strike price to consider in the plot.",
+    )
+    parser.add_argument(
+        "-M",
+        "--max",
+        dest="max_sp",
+        type=check_non_negative,
+        default=-1,
+        help="maximum strike price to consider in the plot.",
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        df_calls, _, _ = get_calls_puts_maxpain(
+            op_calls, pd.DataFrame(), last_adj_close_price
+        )
+
+        if ns_parser.min_sp == -1:
+            min_strike = np.percentile(df_calls["strike"], 25)
+        else:
+            min_strike = ns_parser.min_sp
+
+        if ns_parser.max_sp == -1:
+            max_strike = np.percentile(df_calls["strike"], 75)
+        else:
+            max_strike = ns_parser.max_sp
+
```

Comment:
Maybe default to a percentage of the current stock price instead of middle 50%. The percentile can be pretty large still.

Reply:
Yea, you're right. That's a good point. I used 0.75 and 1.25, let me know if that is good enough :)